src/clarity-angular/data/datagrid/providers/selection.ts

```diff
@@ -154,11 +167,33 @@ export class Selection {
     /**
      * Checks if an item is currently selected
      */
-    public isSelected(item: any): boolean {
-        if (this._selectionType === SelectionType.Single) {
-            return this.currentSingle === item;
-        } else if (this._selectionType === SelectionType.Multi) {
-            return this.current.indexOf(item) >= 0;
+    public isSelected(item: any, rowIndex?: number): boolean {
```

Comment:
We couldn't do simple equality checks on the selection array if we have trackBy values in the selection property. `isSelected` always sends the item, not the reference, so we have to compute its reference and compare. Server driven grids cannot compare equality otherwise.